pypuppetdb/api/__init__.py

```diff
@@ -398,8 +404,14 @@ def nodes(self, unreported=2, with_status=False, **kwargs):
                         node['status'] = 'noop'
                     if status['failures'] > 0:
                         node['status'] = 'failed'
+                elif 'latest_report_status' in node:
+                    if node['latest_report_status'] == 'failed':
+                        node['status'] = 'failed'
                 else:
-                    node['status'] = 'unchanged'
+                    reports = [r for r in latest_reports
+                               if r.node == node['certname']]
```

Comment:
Since the ``reports()`` function yields a generator of results this cannot work in non-development environments. Please read [this StackOverflow question](http://stackoverflow.com/questions/231767/what-does-the-yield-keyword-do-in-python) that effectively explains iterables and generators.